File: src/lessons/welcome/bat/bool2/FizzBuzzEntity.java
package lessons.welcome.bat.bool2;

import static plm.core.ValueSerializer.deserialize;
import static plm.core.ValueSerializer.serialize;

import plm.universe.bat.BatEntity;

public class FizzBuzzEntity extends BatEntity {

  public void run()
  {
    int count = getTestCount();
    for (int i = 0; i < count; i++) {
      Object[] param = (Object[])deserialize(getTest(i));
      setTestResult(i, serialize(fizzBuzz((Integer)param[0])));
    }
  }

  /* BEGIN TEMPLATE */
  String fizzBuzz(int a)
  {
    /* BEGIN SOLUTION */
    if (a % 5 == 0 && a % 3 == 0)
      return "Fizz Buzz";
    else if (a % 5 == 0)
      return "Buzz";
    else if (a % 3 == 0)
      return "Fizz";
    return "" + a;
    /* END SOLUTION */
  }
  /* END TEMPLATE */
}

File: src/lessons/welcome/bat/bool2/FizzBuzzEntity.scala
package lessons.welcome.bat.bool2

import plm.universe.bat.BatEntity
import plm.universe.bat.BatTest

class FizzBuzzEntity extends BatEntity {

    override def run() {
      import plm.core.ValueSerializer._

      val count = getTestCount()
      for (i <- 0 to count -1) {
        val param = deserialize(getTest(i)).asInstanceOf[Array[Object]]
      	setTestResult(i, serialize(fizzBuzz(param(0).asInstanceOf[Int]) ))
	  }
	}

	/* BEGIN TEMPLATE */
	def fizzBuzz(a:Int):String = {
		/* BEGIN SOLUTION */
		(a%5, a%3) match {	    case (0,0) => return "Fizz Buzz"
		    case (_,0) => return "Fizz"
	 	    case (0,_) => return "Buzz"
		    case _     => return ""+a
	    }
		/* END SOLUTION */
	}
	/* END TEMPLATE */
}
